sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4320,33 +4289,48 @@ def _check_existing_cluster(
+            status_before_refresh = record['status']
 
         prev_cluster_status, handle = (status_before_refresh,
                                        handle_before_refresh)
 
         if not dryrun:
-            prev_cluster_status, handle = (
-                backend_utils.refresh_cluster_status_handle(
-                    cluster_name,
-                    # We force refresh for the init status to determine the
-                    # actual state of a previous cluster in INIT state.
-                    #
-                    # This is important for the case, where an existing cluster
-                    # is transitioned into INIT state due to key interruption
-                    # during launching, with the following steps:
-                    # (1) launch, after answering prompt immediately ctrl-c;
-                    # (2) launch again.
-                    # If we don't refresh the state of the cluster and reset it
-                    # back to STOPPED, our failover logic will consider it as an
-                    # abnormal cluster after hitting resources capacity limit on
-                    # the cloud, and will start failover. This is not desired,
-                    # because the user may want to keep the data on the disk of
-                    # that cluster.
-                    force_refresh_statuses={status_lib.ClusterStatus.INIT},
-                    acquire_per_cluster_status_lock=False,
-                ))
+            record = backend_utils.refresh_cluster_record(
+                cluster_name,
+                # We force refresh for the init status to determine the
+                # actual state of a previous cluster in INIT state.
+                #
+                # This is important for the case, where an existing cluster
+                # is transitioned into INIT state due to key interruption
+                # during launching, with the following steps:
+                # (1) launch, after answering prompt immediately ctrl-c;
+                # (2) launch again.
+                # If we don't refresh the state of the cluster and reset it
+                # back to STOPPED, our failover logic will consider it as an
+                # abnormal cluster after hitting resources capacity limit on
+                # the cloud, and will start failover. This is not desired,
+                # because the user may want to keep the data on the disk of
+                # that cluster.
+                force_refresh_statuses={status_lib.ClusterStatus.INIT},
+                acquire_per_cluster_status_lock=False,
+            )
+            if record is not None:
+                prev_cluster_status = record['status']
+                handle = record['handle']
+            else:
+                prev_cluster_status = None
+                handle = None
+        # We should check the cluster_ever_up after refresh, as the cluster
+        # may be terminated, and the cluster_ever_up should be set to False.
```

Comment:
Not sure what 
```
as the cluster
        # may be terminated, and the cluster_ever_up should be set to False.
```
means?

Reply:
Does it mean: cluster was INIT -> terminated on console (or through other means) -> here, we do refresh, and record will become None -> should consider cluster_ever_up = False?